### PR TITLE
Feat/3525 farming project page

### PR DESCRIPTION
**Pull request by schwepps**
Version with migration in portfolio only

### CODE REVIEW

File: app/root.tsx [hunk 1]
```diff
@@ -127,15 +128,25 @@ type Notification = {
...
+};
 
 export function useNotifications() {
   return useOutletContext<ContextType>();
 }
 
 export function ErrorBoundary() {
   const error = useRouteError();
-
+  console.log(error);
```
Comment (MartianGreed) [inline app/root.tsx:149]: it's better to console.error when this is an error

File: app/utils/constant.ts [hunk 2]
```diff
@@ -1 +1,2 @@
-export const GRAMS_PER_TON = 1000000;
\ No newline at end of file
+export const GRAMS_PER_TON = 1000000;
+export const UINT256_DECIMALS = Math.pow(10, 6);
```
Comment (MartianGreed) [inline app/utils/constant.ts:2]: this can be confusing, is it referring to ERC3525_VALUE_DECIMALS or just UINT256_DECIMALS ?

**Review by MartianGreed — CHANGES_REQUESTED** [review]
LAGTM

File: app/utils/constant.ts (hunk 2 above)
Comment (schwepps) [reply app/utils/constant.ts:2]: Currently it's just used in a calldata with UINT256 parameters.
`calldata: tokens.length === 0 ? [amount * UINT256_DECIMALS, 0] : [parseInt(tokens[tokens.length - 1].token_id), 0, amount * UINT256_DECIMALS, 0]`

**Review by MartianGreed — APPROVED** [review]
LGTM

**Pull request merged**